src/workloads/execution/SSBColumnStoreIndex.yml

```diff
@@ -0,0 +1,595 @@
+          "localField": "l_suppkey",
+          "foreignField": "s_suppkey",
+          "as": "supplier"
+        }
+      },
+      {"$addFields": {"supplier": {"$first": "$supplier"}}},
+      {"$match": {"supplier.nation.region.r_name": "AMERICA"}},
+      {
+        "$lookup": {
+          "from": "customer",
+          "localField": "order.o_custkey",
+          "foreignField": "c_custkey",
+          "as": "customer"
+        }
+      },
+      {"$addFields": {"customer": {"$first": "$customer"}}},
+      {"$match": {"customer.nation.region.r_name": "AMERICA"}},
+      {
+        "$group": {
+          "_id": {
+            "order_year": {"$year": "$order.o_orderdate"},
+            "c_nation": "$customer.nation.n_name"
+          },
+          "revenue": {
+            "$sum": {
+              "$subtract": [
+                "$l_extendedprice",
+                {"$multiply": ["$l_extendedprice", "$l_discount"]}
+              ]
+            }
+          }
+        }
+      },
+      {"$addFields": {"order_year": "$_id.order_year", "customer_nation": "$_id.c_nation"} }
+    ]
+  cursor: {}
+
+Q4_3Query: &Q4_3Query
+  aggregate: *Collection
+  pipeline:
+    [
+      { $limit: 10000 },
+      {
+        "$match": {
+          "part.p_mfgr": "Manufacturer#1",
+          "order.o_orderdate": {
+            "$gte": ISODate("1997-01-01T00:00:00.000Z"),
+            "$lt": ISODate("1999-01-01T00:00:00.000Z")
+          }
+        }
+      },
+      {
+        "$lookup": {
+          "from": "supplier",
+          "localField": "l_suppkey",
+          "foreignField": "s_suppkey",
+          "as": "supplier"
+        }
+      },
+      {"$addFields": {"supplier": {"$first": "$supplier"}}},
+      {"$match": {"supplier.nation.n_name": "UNITED STATES"}},
+      {
+        "$lookup": {
+          "from": "customer",
+          "localField": "order.o_custkey",
+          "foreignField": "c_custkey",
+          "as": "customer"
+        }
+      },
+      {"$addFields": {"customer": {"$first": "$customer"}}},
+      {"$match": {"customer.nation.region.r_name": "AMERICA"}},
+      {
+        "$group": {
+          "_id": {
+            "order_year": {"$year": "$order.o_orderdate"},
+            "c_nation": "$customer.nation.n_name",
+            "p_brand": "$part.p_brand"
+          },
+          "revenue": {
+            "$sum": {
+              "$subtract": [
+                "$l_extendedprice",
+                {"$multiply": ["$l_extendedprice", "$l_discount"]}
+              ]
+            }
+          }
+        }
+      },
+      {
+        "$addFields": {
+          "order_year": "$_id.order_year",
+          "customer_nation": "$_id.c_nation",
+          "part_brand": "$_id.p_brand"
+        }
+      },
+      {"$sort": {"order_year": 1, "customer_nation": 1, "part_brand": 1}}
+    ]
+  cursor: {}
+
+# Defines test operations.
+# Here we use explain with verbosity = execustionStats instead of using plain aggregate command
```

Comment:
David Storch recently updated the CRUD actor to support aggregation (https://github.com/mongodb/genny/commit/bba39f66d82d02bfc9c064e8b3ec688e3a76f46a). Could it be used now instead of the explain-based workaround?

Reply:
Resolved in 9d03739054cac7e76273af8b1f82a7d0005765e7